src/components/pod/pod.test.js

```diff
@@ -13,22 +59,236 @@ const PodWithSoftDelete = () => (
+        podBlock().realHover().and("have.css", "background-color", color);
+        podEdit().should("have.css", "background-color", color);
+      }
+    );
+
+    it.each([
+      [true, 1349, "rgba(0, 0, 0, 0)"],
+      [false, 77, "rgb(0, 103, 56)"],
+    ])(
+      "should check the width value when internalEditButton is %s for Pod component",
+      (boolVal, width, color) => {
+        CypressMountWithProviders(
+          <PodComponent internalEditButton={boolVal} />
+        );
+        divPodComponent().then(($el) => {
+          useJQueryCssValueAndAssert($el, "width", width);
+        });
+        podEdit().realHover().and("have.css", "background-color", color);
+        divPodComponent().should(
+          "have.css",
+          "background-color",
+          "rgb(217, 224, 228)"
+        );
+      }
+    );
+
+    it.each([100, 200, 300])(
+      "should check when internalEditButton is %s for Pod component",
+      (height) => {
+        CypressMountWithProviders(<PodComponent height={height} />);
+        podComponent().should("have.css", "height", `${height}px`);
+      }
+    );
+
+    describe("should render Pod component for event tests", () => {
+      let callback;
+      beforeEach(() => {
+        callback = cy.stub();
+      });
+
+      it("should call onDelete callback when a click event is triggered for Pod component", () => {
+        CypressMountWithProviders(<PodComponent onDelete={callback} />);
+        podDelete()
+          .click()
+          .then(() => {
+            // eslint-disable-next-line no-unused-expressions
+            expect(callback).to.have.been.calledOnce;
+          });
+      });
+
+      it("should call onUndo callback when a click event is triggered for Pod component", () => {
+        CypressMountWithProviders(
+          <SoftDeletePodComponent softDelete onUndo={callback} />
+        );
+        podUndo()
+          .click()
+          .then(() => {
+            // eslint-disable-next-line no-unused-expressions
+            expect(callback).to.have.been.calledOnce;
+          });
+      });
+    });
+
+    describe("When softDelete is true", () => {
+      it("renders block with correct background colour", () => {
+        const blockBackgroundColor = "rgb(230, 235, 237)";
+
+        CypressMountWithProviders(<PodWithSoftDelete />);
+
+        podBlock().should("have.css", "background-color", blockBackgroundColor);
+      });
+
+      it("renders text of children with correct colours", () => {
+        const childrenColor = "rgba(0, 0, 0, 0.65)";
 
-      CypressMountWithProviders(<PodWithSoftDelete />);
+        CypressMountWithProviders(<PodWithSoftDelete />);
 
-      cy.contains("Content").should("have.css", "color", childrenColor);
-      cy.contains("More content").should("have.css", "color", childrenColor);
+        cy.contains("Content").should("have.css", "color", childrenColor);
+        cy.contains("More content").should("have.css", "color", childrenColor);
+      });
```

Comment:
You could probably remove the `PodWithSoftDelete` component here as you could achieve the same result using `SoftDeletePodComponent` component instead.

Reply:
Both the custom components are different as we are passing props in SoftDeletePodComponent.